syntax = "proto3";

message UserProfile {
  uint64 user_id = 1;
}

message AdInfo {
  uint64 ad_id = 1;
  float score = 2;
}

message StrategyRequest {
  int64 logid = 1;
  string graph = 2;
  UserProfile user_profile = 3;
  repeated AdInfo ad_infos = 4;
}

message StrategyResponse {
  enum Status {
    OK = 0;
    ERROR = 1;
  }
  Status status = 1;
  repeated AdInfo ad_infos = 2;
}

service StrategyService {
  rpc Rank (StrategyRequest) returns (StrategyResponse) { }
}
